nixos/modules/services/monitoring/kthxbye.nix

```diff
@@ -0,0 +1,159 @@
+{ config, pkgs, lib, ... }:
+with lib;
+
+let
+  cfg = config.services.kthxbye;
+in
+
+{
+  options.services.kthxbye = {
+    enable = mkEnableOption (mdDoc "kthxbye alert acknowledgement management daemon");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.kthxbye;
+      defaultText = literalExpression "pkgs.kthxbye";
+      description = mdDoc ''
+        The kthxbye package that should be used.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Open ports needed for the daemon to function.
+      '';
+      example = true;
+    };
+
+    extraOptions = mkOption {
+      type = with types; listOf str;
+      default = [];
+      description = mdDoc ''
+        Extra command line options.
+      '';
+      example = literalExpression ''
+        [
+          "-extend-with-prefix 'ACK!'"
```

Comment:
This isn't the best example because you already have an `extendWithPrefix` option which does this.

Reply:
kthxbye doesn't have any unique command-line options I can use as an example since they all can be set in the config file. I left this option in so that if anyone decides to override the package with a newer version, they'll be able to use the options which aren't available in the nix config.

Reply:
Config file? I thought this application was only configurable via command line options. If a configuration file is possible this is **highly** desirable over using command line options. `nix` is much better with configuration files.

Reply:
Edited above comment, this application is only configurable using command-line options. That said, the upstream author did describe this as a small daemon so they probably don't think it's worth using config files yet.

Reply:
Ah I see. Yes, seems fine to leave this as is then. Ignore my comment. Thanks!